spring-cloud-gateway-core/src/main/java/org/springframework/cloud/gateway/route/Route.java

```diff
@@ -50,14 +51,17 @@
 
 	private final List<GatewayFilter> gatewayFilters;
 
+	private final RouteMetadata metadata;
```

Comment:
I'd prefer this to be just a Map.

Reply:
I made it as a sepate class, to be able to add "typed" builder methods for different pojos. And use it in Java route builders.
Is my undenderstanding is correct that the right way for it, is to create a custom RouteBuilder ancestor for it? And that's why this class is extra one?

Reply:
I don't see any reason for the custom class. The whole point is not to have anything "typed".

Reply:
the reason is to have the possibility to make a few typed getter and setters for the features like timeouts or smth similar.
I usually prefer as typed code as possible but probably in such projects, it is not the way to go